src/Library/demos/Text Input/main.blp

```diff
@@ -0,0 +1,309 @@
+          primary-icon-activatable: true;
+          primary-icon-tooltip-text: "Click on me!";
+          secondary-icon-name: "bell-symbolic";
+          secondary-icon-tooltip-text: "No click on me!";
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Visiblity";
+          margin-bottom: 10;
+        }
+
+        Entry visibility_entry {
+          visibility: false;
+          secondary-icon-name: "eye-open-negative-filled-symbolic";
+          secondary-icon-activatable: true;
+          buffer: bind entry.buffer;
+        }
+      }
+    }
+
+    Box {
+      halign: center;
+      margin-top: 20;
+      margin-bottom: 20;
+
+      LinkButton{
+        label: "Tutorial";
+        uri: "https://developer.gnome.org/documentation/tutorials/beginners/components/entry.html";
+      }
+
+      LinkButton{
+        label: "API Reference";
+        uri: "https://docs.gtk.org/gtk4/class.Entry.html";
+      }
+    }
+
+    Separator {
+      margin-bottom: 20;
+    }
+
+    Label {
+      label: "Search Entry";
+      halign: start;
+
+      styles [
+        "heading",
+      ]
+    }
+
+    Label {
+      label: "An entry widget that has been tailored for searching.";
+      margin-top: 10;
+      margin-bottom: 10;
+      halign: start;
+
+      styles [
+        "dim-label",
+      ]
+    }
+
+    SearchEntry search_entry {
+      margin-bottom: 10;
+      placeholder-text: "Search for items...";
+    }
+
+    ListBox items {
+      margin-bottom: 10;
+
+      styles [
+        "boxed-list",
+      ]
+
+      ListBoxRow {
+        Label {
+          label: "Apples";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Oranges";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Bananas";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Strawberries";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Pineapples";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Blueberries";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+    }
+
+    LinkButton {
+      label: "API Reference";
+      uri: "https://docs.gtk.org/gtk4/class.SearchEntry.html";
+      margin-top: 20;
+      margin-bottom: 20;
+    }
+
+    Separator {
+      margin-bottom: 20;
+    }
+
+    Label {
+      label: "Password Entry";
+      halign: start;
+
+      styles [
+        "heading",
+      ]
+    }
+
+    Label {
+      label: "An entry with dedicated control for entering secrets and passwords.";
+      margin-top: 10;
```

Comment:
can please use multiple of 6? it's the GNOME guideline

so `12` here would work well

Reply:
Okay I'll change it everywhere 👍🏾